### DISCUSSION TITLE
Parenthesis around functions?

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#1764] [Polls]
Do you like that PRQL doesn't use any parenthesis around function declarations and functions calls or would you prefer parenthesis?

## Presently in PRQL
Declaring a function in PRQL:

    func fahrenheit_from_celsius temp -> temp * 1.8 + 32

Calling a function in PRQL:

    fahrenheit_from_celsius 123

## But maybe?

Declaring a function in PRQL:

    func fahrenheit_from_celsius(temp) -> temp * 1.8 + 32

Calling a function in PRQL:

    fahrenheit_from_celsius(123)

**Comment by max-sixty** [comment]
I appreciate the interest @vanillajonathan , thanks for the burst of enthusiasm on the issue tracker!

For folks who are interested in how we're thinking about syntax, we touch on function syntax briefly within the switch & match syntax discussion at https://github.com/PRQL/prql/issues/1286. I'm sure there's also lots of discussion deep in the archives about parentheses / spaces / etc.

---

@vanillajonathan to offer some guidance about making an impact on PRQL: a decision like this requires lots of context — both PRQL's and other languages' designs — and aesthetics are only a small part of that decision. Most of what we'd get from a poll like this would be folks' views on the aesthetics, and so it's unlikely to make an impact on the language.

I'd encourage you to find issues where you can have an impact immediately, gain some context about the language as you do that, and then address some bigger topics like this. 

Some of your issues today have been excellent, I appreciate them. I hope it's reasonable to write this, and I hope you grow your involvement with PRQL.

---

That said, let's leave this open — maybe it will inspire someone to write something thoughtful about the options for this syntax. (It would need to be quite thoughtful and considered to have an impact...)
